[Guide updates] Custom EST certificate authority and Fleet API to request a certificate

noahtalerman opened the pull request:
Guide updates for the following user story:

- https://github.com/fleetdm/fleet/issues/32110

noahtalerman committed:
Update guide

mike-j-thomas commented on articles/connect-end-user-to-wifi-with-certificate.md:
```diff
@@ -195,28 +194,43 @@ When the profile is delivered to your hosts, Fleet will replace the variables. I
-  - For `Challenge`, use`$FLEET_VAR_CUSTOM_SCEP_CHALLENGE_<CA_NAME>`. 
-  - For `URL`, use `$FLEET_VAR_CUSTOM_SCEP_PROXY_URL_<CA_NAME>`, and make sure to add `$FLEET_VAR_SCEP_RENEWAL_ID` to `CN`.
+  - For `Challenge`, use`$FLEET_VAR_SMALLSTEP_SCEP_CHALLENGE_<CA_NAME>`. 
+  - For, `URL`, use `$FLEET_VAR_SMALLSTEP_SCEP_PROXY_URL_<CA_NAME>`, and make sure to add `$FLEET_VAR_SCEP_RENEWAL_ID` to `OU`.
```
```suggestion
  - For `URL`, use `$FLEET_VAR_SMALLSTEP_SCEP_PROXY_URL_<CA_NAME>`, and make sure to add `$FLEET_VAR_SCEP_RENEWAL_ID` to `OU`.
```

mike-j-thomas commented on articles/connect-end-user-to-wifi-with-certificate.md:
```diff
@@ -286,43 +300,117 @@ When the profile is delivered to your hosts, Fleet will replace the variables. I
+This custom script will create a certificate signing request (CSR) and make a request to Fleet's "Request certificate" API endpoint.
 
-2. In the modal, select **Smallstep** from the dropdown and enter a name for your certificate authority (CA). Best practice is all caps snake case (for example, "WIFI_AUTHENTICATION"). This name is used later as a variable name in a configuration profile.
+1. Create an API-only user with the global maintainer role. Learn more how to create an API-only user in the [API-only user guide](https://fleetdm.com/guides/fleetctl#create-api-only-user).
```
```suggestion
1. Create an API-only user with the global maintainer role. Learn how to create an API-only user in the [API-only user guide](https://fleetdm.com/guides/fleetctl#create-api-only-user).
```

mike-j-thomas commented on articles/connect-end-user-to-wifi-with-certificate.md:
````diff
@@ -471,8 +560,50 @@ CLIENT_ID="<OAuth-IdP-client-ID>"
+fetch_cert -ca <EST-CA-ID> -fleeturl "<Fleet-server-URL>" -csr CustomerUserNetworkAccess.csr -out /opt/company/certificate.pem
+```
+
+This script assumes that your company installs a custom Company Portal app or something similar at `/opt/company`, gathers the user's IdP session information, uses username and a password to protect the private key from `/opt/company/userinfo`, and installs that the certificate in `/opt/company`. You will want to modify it to match your company's requirements.
````
Some typos in here.

mike-j-thomas reviewed (changes requested):
A few typos and a suggested rewrite of a paragraph that tripped me up while reading.

mike-j-thomas commented:
@noahtalerman, might not be an issue, but I noticed [this PR](https://github.com/fleetdm/fleet/pull/35445/files) is also modifying the same article.

noahtalerman committed:
Update articles/connect-end-user-to-wifi-with-certificate.md

Co-authored-by: Mike Thomas <78363703+mike-j-thomas@users.noreply.github.com>

noahtalerman committed:
Update articles/connect-end-user-to-wifi-with-certificate.md

Co-authored-by: Mike Thomas <78363703+mike-j-thomas@users.noreply.github.com>

noahtalerman committed:
Update connect-end-user-to-wifi-with-certificate.md

noahtalerman committed:
Update articles/connect-end-user-to-wifi-with-certificate.md

Co-authored-by: Mike Thomas <78363703+mike-j-thomas@users.noreply.github.com>

noahtalerman committed:
Update script deployment instructions for Linux hosts

Revised instructions for creating a custom script to deploy certificates automatically to Linux hosts during enrollment.

noahtalerman committed:
Update articles/connect-end-user-to-wifi-with-certificate.md

Co-authored-by: Mike Thomas <78363703+mike-j-thomas@users.noreply.github.com>

noahtalerman committed:
Update articles/connect-end-user-to-wifi-with-certificate.md

Co-authored-by: Mike Thomas <78363703+mike-j-thomas@users.noreply.github.com>
